### ISSUE TITLE
Lobby rules - breaking through the lobby with a guest identity

### ISSUE THREAD

**Issue by paulnearney** [microsoftgraph/microsoft-graph-comms-samples#353]
I'm trying to understand if it's possible to get a bot joining a meeting as a guest, and have it bypass the lobby if the lobby controls are set to anything but "Everyone". I've been testing all combinations of settings that could influence this, but none appear to make any difference:

- Using `Calls.JoinGroupCall.All` vs `Calls.JoinGroupCallAsGuest.All` on the Azure AD app reg
- Setting the `JoinMeetingParameters.AllowGuestToBypassLobby` setting to true, false, and leaving as null
- Joining with and without a `JoinMeetingParameters.GuestIdentity`

I'm setting the GuestIdentity prior to joining, like so:
```
    var joinParams = new JoinMeetingParameters(chatInfo, meetingInfo, mediaSession)
    {
        TenantId = meetingInfo.Organizer.User.GetTenantId(),
        GuestIdentity = new Identity { DisplayName = "Paulie", Id = Guid.NewGuid().ToString() },
        AllowGuestToBypassLobby = true
    };
```

Regardless of which combination of settings I use, it seems that the guest user can only automatically get through the lobby if the lobby settings are set to "Everyone", as shown in the matrix below (this matrix is the same regardless of whether the Azure AD app has the `Calls.JoinGroupCall.All` or the `Calls.JoinGroupCallAsGuest.All` permission):

  | Guest, AGTBL = false | Guest, AGTBL = true | App, AGTBL = false | App, AGTBL = true
-- | -- | -- | -- | --
Only Me | No, 75 sec timeout | No, 75 sec timeout | Yes | Yes
Org | No, 75 sec timeout | No, 75 sec timeout | Yes | Yes
Org + Trusted | No, 75 sec timeout | No, 75 sec timeout | Yes | Yes
Everyone | Yes | Yes | Yes | Yes

Where I refer to the "75 sec timeout", this means that on attempting to join the call, the call state eventually changes to `Terminated` with a `ResultInfo` of:

```
    AdditionalData: Count = 0
    Code: 410
    Message: "An answer for the start incoming negotiation request was not received within 75 seconds.. DiagCode: 410#319009"
    ODataType: "microsoft.graph.resultInfo"
    Subcode: 9999
```

This doesn't seem right to me. Surely the Azure AD permissions that are selected, and/or the `AllowGuestToBypassLobby` setting should make some difference to guest lobby bypass?

My ideal is to be able to get the guest user to join the meeting regardless of the lobby settings - is this possible? if not, what's the lowest level of lobby settingsthat would let the guest get in to the meeting , and how would I achieve it?

**Comment by paulnearney** [comment]
Is there any update on this please?

**Comment by paulnearney** [comment]
Is there any update on this please?

**Comment by 1fabi0** [comment]
In the HueBot Sample it's also possible to join as Guest maybe you should check it out https://github.com/microsoftgraph/microsoft-graph-comms-samples/blob/a4c6c6c8ae96cbd0125851dcc64278f9d02328cb/Samples/V1.0Samples/LocalMediaSamples/HueBot/HueBot/Bot/Bot.cs#L146

**Comment by paulnearney** [comment]
Thanks @1fabi0, but the question isn't around how to join as a guest, but the lobby rules that apply for guest applications, and whether they can be bypassed.

**Comment by 1fabi0** [comment]
Did you already try to enable Lobby bypass for the Teams meeting?

**Comment by paulnearney** [comment]
Yes. As per the question, the only way to get a guest to bypass the lobby is when lobby bypass is set to "Everyone". AD permissions and the AllowGuestToBypassLobby seem to imply it should be possible to change this behaviour, but they don't seem to have any effect.

**Comment by 1fabi0** [comment]
Yes but I hope you understand the point of microsoft if you want to bypass lobby join as application, maybe it's possible for you to join as application that accepts the guests but why would you then want a guest

**Comment by paulnearney** [comment]
Joining as a guest is required in our use case as it allows us to control the display name, and join multiple times to the same call. Unfortunately there's no ability for an application to admit users/guest apps. The best we can do is to have the application change lobby settings on the fly but this isn't acceptable to the customer.

The AD permissions (Calls.JoinGroupCall.All / Calls.JoinGroupCallAsGuest.All) and the JoinMeetingParameters.AllowGuestToBypassLobby setting imply that they have some control over lobby bypass for guests, but in my testing, they don't seem to have any effect at all.

It would be great to get clarity from Microsoft on what effects these settings should have, and whether there are any issues with them currently.
